src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Block.cs

```diff
@@ -20,26 +19,49 @@ public override BoundNode VisitBlock(BoundBlock node)
             }
 
             var builder = ArrayBuilder<BoundStatement>.GetInstance();
-            VisitStatementSubList(builder, node.Statements);
-
-            var additionalLocals = TemporaryArray<LocalSymbol>.Empty;
+            var previousLocals = _additionalLocals;
```

Comment:
A comment about what we are trying to do might be appropriate, e.g. when `_additionalLocals` is null here, it means we are entering the outermost block of the current function. Symbols inserted to `_additionalLocals` by child statements should be put in this block.